xdsl/ir.py

```diff
@@ -544,6 +534,27 @@ def irdl_definition(cls) -> OpDef:
         ...
 
 
+@dataclass
```

Comment:
I'm not sure if this would be the place to add this operation.
Maybe we should create a file like `ir_helper.py`? What do you think?
I would say that `ir.py` should only contains the necessary IR constructs.

Reply:
Isn't an OPeration an IR Construct?

Reply:
> Maybe we should create a file like ir_helper.py
It could also be just operation.py

Reply:
I think this conversation is not yet resolved. AFAIU, this kind of class is like providing a special kind of operation that users can use out of the box instead of having to define these kinds of parent classes themselves. Therefore, Mathieu would prefer to use it out of this file, that should mainly concern IR constructs and not such helper constructs.

Reply:
Yes, could we move this to some other file?
Something like `ir_helper.py`, or `definition_utils.py`.
I would really like to have only the strict necessary in `ir.py`, to make it easier to understand the core structure of xdsl.